compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -843,6 +876,59 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         Ok(())
     }
 
+    /// Pops a stack frame from the stack and returns some information about it.
+    ///
+    /// This also deallocates locals, if necessary.
+    ///
+    /// [`M::before_stack_pop`] should be called before calling this function.
+    /// [`M::after_stack_pop`] is called by this function automatically.
+    ///
+    /// [`M::before_stack_pop`]: Machine::before_stack_pop
+    /// [`M::after_stack_pop`]: Machine::after_stack_pop
+    pub fn pop_stack_frame(
+        &mut self,
+        unwinding: bool,
+    ) -> InterpResult<'tcx, StackPop<'tcx, M::Provenance>> {
+        let cleanup = self.cleanup_current_frame_locals()?;
+
+        let frame =
+            self.stack_mut().pop().expect("tried to pop a stack frame, but there were none");
+
+        let target = frame.return_to_block;
+        let destination = frame.return_place.clone();
```

Comment:
**nitpick:** This clone seems to be unnecessary and could be moved onto the else branch.

Reply:
I don't think it can be moved into the else branch? It could be moved into the "`if`" branch (where `frame` is moved), but this seems unnecessary -- `MPLace` has no allocations, so `clone` is basically `memcpy`, same as a move. Here is seems that doing a conditional clone/move would only complicate code without benefit.